simulators/portal-interop/src/main.rs

```diff
@@ -922,3 +935,92 @@ dyn_async! {
         }
     }
 }
+
+dyn_async! {
+    async fn test_gossip_two_nodes<'a> (client_a: Client, client_b: Client) {
+        // connect clients
+        let client_b_enr = match client_b.rpc.node_info().await {
+            Ok(node_info) => node_info.enr,
+            Err(err) => {
+                panic!("Error getting node info: {err:?}");
+            }
+        };
+        match HistoryNetworkApiClient::add_enr(&client_a.rpc, client_b_enr.clone()).await {
+            Ok(response) => match response {
+                true => (),
+                false => panic!("AddEnr expected to get true and instead got false")
+            },
+            Err(err) => panic!("{}", &err.to_string()),
+        }
+
+        // With default node settings nodes should be storing all content
+        let values = std::fs::read_to_string("./test-data/test_data_collection_of_forks_blocks.json")
+            .expect("cannot find test asset");
+        let values: Value = serde_json::from_str(&values).unwrap();
+
+        for value in values.as_array().unwrap() {
+            let content_key: HistoryContentKey =
+                serde_json::from_value(value.get("content_key").unwrap().clone()).unwrap();
+            let content_value: HistoryContentValue =
+                serde_json::from_value(value.get("content_value").unwrap().clone()).unwrap();
+
+            match client_a.rpc.gossip(content_key.clone(), content_value.clone()).await {
+                Ok(nodes_gossiped_to) => {
+                   if nodes_gossiped_to != 1 {
+                        panic!("We expected to gossip to 1 node instead we gossiped to: {nodes_gossiped_to}");
+                    }
+                }
+                Err(err) => {
+                    panic!("Unable to get received content: {err:?}");
+                }
+            }
+
+            if let HistoryContentKey::BlockHeaderWithProof(_) = content_key {
+                tokio::time::sleep(Duration::from_secs(1)).await;
+            }
+        }
+
+        // wait 8 seconds for data to propagate
+        // This value is determined by how long the sleeps are in the gossip code
+        // So we may lower this or remove it depending on what we find.
+        tokio::time::sleep(Duration::from_secs(8)).await;
+
+        let comments = vec!["1 header", "1 block body", "100 header",
+            "100 block body", "7000000 header", "7000000 block body",
+            "7000000 receipt", "15600000 (post-merge) header", "15600000 (post-merge) block body", "15600000 (post-merge) receipt",
+            "17510000 (post-shanghai) header", "17510000 (post-shanghai) block body", "17510000 (post-shanghai) receipt"];
+
+        let mut results = vec![];
+        for client in &vec![client_a, client_b] {
```

Comment:
In Fluffy we recently added in not to gossip back the content to its source node.

This is not in the specification currently, but does make sense (I can suggest to add it to the specification as a `SHOULD` or `MAY`). In any case, I don't think it makes sense to add this into this test, as for sure it should be an allowed optimization in the network.

So I think we should only test the receiving client here for content.

Reply:
BTW here PR to address that: https://github.com/ethereum/portal-network-specs/pull/229

Reply:
When is recently? because If I recall I tested fluffy and it gossiped back to A when I originally made this PR.

Reply:
> When is recently? because If I recall I tested fluffy and it gossiped back to A when I originally made this PR.

Recently is 4th September: https://github.com/status-im/nimbus-eth1/commit/69a1a988b0a7b072d4b2ed7a3cf336e57a2e7b87

So if it is still gossiping back after this commit, then we have a bug.

Reply:
``"pass":false,"details":"Client A: [\"Error content for block 17510000 (post-shanghai) block body was absent\"] Client B: [\"Error content for block 17510000 (post-shanghai) block body was absent\"]"``

It looks like you have a bug? I just pulled the docker image (unless my build system is broken). The error meaning Fluffy Node A is storing everything but post-shanghai block body (which is a known bug).

If you want I can wipe all docker images/volumns off my drive and try again let me know.

Because if we call Gossip from node A -> B from what you are saying I would expect A to not have any of this content.

Reply:
Thanks, that's strange indeed. Anyhow, the test can be altered imo so that just client b gets tested for content.

We will debug in case A still has the content also.

Reply:
Ok I resolved your PR comment